### PR TITLE
feat: #37 & #39 - Implement the api with filters

### LINKED ISSUE #37: Implement The API Get Function To Create Features In The APP
### Explanation:
In the `actions` folder, there is a function that does the request to one public art API and returns the data got. This function is partially implemented in the `/artworks/` view.

Actually this function is getting 10.000 artworks and showing it simulating pagination, we need to replace this logic for a real pagination logic, using `limit` to get just 16 artworks that we need, and using `offset` to move about the pages using the paginator.  

We need too do work the search feature, this work adding a `q` param to the API query. 

### Screenshots
imagen

### Checklist
 This issue can be closed when the following tasks are completed:

- [ ] It shows 16 products by page.
- [ ] It works the pagination.
- [ ] It works the search feature.
- [ ] There is an easy way to change the limit of products by page.
 
#### Related issues:
- #35 

### References:
- [API Documentation](https://openaccess-api.clevelandart.org/#api-endpoints)

### PR DESCRIPTION
**Pull request by Eghinner**
### Description

Below I describe the changes per file

`types.js`

1. I ordered the file and remove unnecessary async in src/actions/types.js

`src/actions/artworks.js`

1. Update `getArtworks` function with params and conditional in template strings for the GET route

`Filters.jsx`

1. The first step was add styles to limit height and scroll hidden when list options from 'Departamentos' or 'Por tipos' are 
deployed
3. Replaced 'category' for 'department' in variables and functions
4. Remove or replace "Por Fecha" references for "Por Tipos"
5. Implement list of types and events like 'department' do it
6. Added functionality to push new `type` and `department` value and not just replace all. This delete `page` key too

`Artworks.js`

1. Remove filterAll function
2. Implement the GET API function with `filters` as argument. `filters` is a object with the URL params
3. Every time `filters` change, the async function is executed updating `artwoks` and `totalArtworks`
4. Remove unnecessary states for pagination and implement `scrollTop` basic
5. Too implement function to set URL params with `page` key

`src/components/ArtworkItem.jsx`

1. I add some temporary tags to see types and department artwork

`src/components/SearchInput.jsx`

1. Added functionality to push new `q` value and not just replace

fixes #37 #39

#### Screenshots

artworks
artworks2

#### Type of change

- [x] New feature (non-breaking change which adds functionality)

### How to test it

1. Go to '/artworks' route
6. In aside section click 'Departamentos' or 'Por tipos' to deploy their respective list
7. Select any option.... and wait a moment
8. Scroll down and navigate with pagination component
9. Search in the main navbar

### Checklist:

- [x] Doesn't break the current code.
- [ ] Passes linters and test, also is building.
- [ ] Doesn't have spelling or grammatical problems.
- [ ] Doesn't have unnecessary comments or debugging code.
- [x] The branch is updated with the last dev branch changes.

### CODE REVIEW

File: src/components/ArtworkItem.jsx [hunk 1]
```diff
@@ -21,6 +21,20 @@ const ArtworkItem = ({ data }) => (
           {data.creation_date}
         </p>
       </div>
+      {
+        // Para el testo
```
Comment (Israel-Laguan) [inline src/components/ArtworkItem.jsx]: Sorry, can you explain here?

File: src/components/Products/Filters.jsx [hunk 2]
```diff
@@ -1,27 +1,50 @@
...
+  selectedFilter,
   setFilter,
+  setSelectedDepartment,
+  setSelectedTypes
 }) => {
+  const { search } = useLocation();
   const { push } = useHistory();
 
+  const updateQueryParams = (key, value) => {
+    const searchParams = new URLSearchParams(search);
```
Comment (Israel-Laguan) [inline src/components/Products/Filters.jsx]: What if is empty? What about to add a default?

File: src/pages/Artworks.js [hunk 3]
```diff
@@ -1,184 +1,144 @@
...
+    const handlePageClick = (event) => {
+        window.scrollTo(0,0)
+        addQuery('page', (event.selected + 1))
+    };
+
+    return ( 
+      <>
+        <MainNav query={queryString.parse(search).q} />
+
+        <div className = "max-w-screen-2xl mx-auto flex flex-col lg:flex-row px-5 sm:px-20 pt-5 lg:pt-16 pb-20" >
```
Comment (Israel-Laguan) [inline src/pages/Artworks.js]: Nit: spaces around className

File: src/pages/Artworks.js [hunk 4]
```diff
@@ -1,184 +1,144 @@
...
           setFilter={setSelectedFilter}
-          currentFilter={selectedFilter}
-        />
-        <div className="w-full lg:w-4/5 space-y-8">
-          {selectedFilter === "none" && (
+          selectedFilter={selectedFilter}
+          setSelectedDepartment={setSelectedDepartment}
+          setSelectedTypes={setSelectedTypes}
+        /> 
+        <div className = "w-full lg:w-4/5 space-y-8" >
```
Comment (Israel-Laguan) [inline src/pages/Artworks.js]: Nit: spaces around className

File: src/pages/Artworks.js [hunk 5]
```diff
@@ -1,184 +1,144 @@
...
-          ) : (
+            : 
             <>
               <div className="grid md:grid-cols-2 xl:grid-cols-4 gap-4 w-full p-0 justify-evenly xl:justify-around">
-                {currentPosts.map((product) => (
+                {
+                  artworks.map((product) => 
                   <ArtworkItem
                     data={product}
                     key={`product-item-${product.id}`}
```
Comment (Israel-Laguan) [inline src/pages/Artworks.js]: please check indentation

File: src/components/Products/Filters.jsx (hunk 2 above)
Comment (Eghinner) [reply src/components/Products/Filters.jsx]: Is about URLSearchParams? Well, if URLSearchParams is empty just create a new object URLSearchParams, then add a specified key/value pair as a new search parameter.
I'm not sure what I should add by default.

File: src/pages/Artworks.js (hunk 3 above)
Comment (Eghinner) [reply src/pages/Artworks.js]: done

File: src/pages/Artworks.js (hunk 4 above)
Comment (Eghinner) [reply src/pages/Artworks.js]: done

File: src/components/ArtworkItem.jsx (hunk 1 above)
Comment (Eghinner) [reply src/components/ArtworkItem.jsx]: A comment that escaped me.
I wanted to leave a section to test the requests

**Review by Israel-Laguan — APPROVED** [review]

File: src/hooks/useSearchParams.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    const updateQueryParams = (key, value) => {
+      searchParams.delete('page');
+      searchParams.set(key, value);
+      push({
+        pathname: 'artworks',
+        search: searchParams.toString()
+      });
+    };
+	
+	return { addQuery, updateQueryParams };
```
Comment (heliomar-pena) [inline src/hooks/useSearchParams.js]: Please check indentation

File: src/actions/artworks.js [hunk 7]
```diff
@@ -21,11 +21,15 @@ export const getArtworkById = async (id, callback) => {
   }
 };
 
-export const getArtworks = async () => {
+export const getArtworks = async ( {department, type, page: skip, q} ) => {
+  const offset = skip ? (parseInt(skip) - 1) * 16 : undefined;
```
Comment (heliomar-pena) [inline src/actions/artworks.js]: You can use the variable that you have in Artworks view to ease to change the quantity of artworks to show 

https://github.com/Azordev/art-marketplace/blob/12b755425a7032dff5d4970028c4f11ed342d4a9/src/pages/Artworks.js#L42

File: src/actions/artworks.js [hunk 8]
```diff
@@ -21,11 +21,15 @@ export const getArtworkById = async (id, callback) => {
...
   try {
-    const res = await authAxios.get(`/artworks?has_image=1&limit=25`);
+    const res = await authAxios
+      .get(`/artworks?has_image=1&limit=16${q ? `&q=${q}` : ''}${department ? `&department=${department}` : ''}${type ? `&type=${type}` : ''}${skip ? `&skip=${offset}` : ''}`);
```
Comment (heliomar-pena) [inline src/actions/artworks.js]: You can use a variable for `limit` variable

Comment (heliomar-pena) [reply src/actions/artworks.js]: You can move all queries strings to variables and just concatenate them here, I think it's more readable. What do you think?

**Review by heliomar-pena — APPROVED** [review]
Good work! Please consider this comments

**Review by angelik0828 — APPROVED** [review]

**Review by Israel-Laguan — APPROVED** [review]
Good work man!

**Review by e-azocar — APPROVED** [review]
It's really good!

**Pull request merged**